### PR TITLE
demo: add support for getting spec from clipboard

### LINKED ISSUE #1251: Add a file selector as well as URL input for the demo
To make it easier to develop an openapi file locally, would it be possible to allow the user to upload a file, as an alternative to entering a URL?

### PR DESCRIPTION
**Pull request by davidjgoss**
image

Adds a button that, when clicked, will try to read the clipboard and import the spec from that.

- Uses the modern [async clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API) (asks the user's permission first)
- Won't be shown in older browsers where it's not supported
- Uses the YAML parser so importing either YAML or JSON will work

Maybe closes #1251 - addresses the use case indirectly.

### CODE REVIEW

**Comment by adamaltman** [comment]
Interesting idea. I think we would need to update that hint text to say they can paste it in too.

**Comment by davidjgoss** [comment]
@adamaltman good point. Maybe something like:

> Enter a spec URL to try, or hit the paste icon to pull from clipboard

(Struggling to find more concise wording.)

Could also look at giving the button a short label like "Import from Clipboard"?

**Comment by davidjgoss** [comment]
@adamaltman take 2:

image

Let me know what you think.

**Comment by davidjgoss** [comment]
Any feedback on the updates here?

**Comment by davidjgoss** [comment]
Given https://github.com/Redocly/redoc/pull/1859, this might be redundant now?
